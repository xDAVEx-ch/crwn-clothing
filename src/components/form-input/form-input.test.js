import React from 'react';

import { shallow } from 'enzyme';

import FormInput from './form-input.component';


describe('FormInput component', () =>{
    let wrapper;
    let mockHandler;

    beforeEach(() =>{
        mockHandler = jest.fn();

        const mockProps = {
            value: 'myemail@gmail.com',
            handleChange: mockHandler,
            label: 'email'
        }

        wrapper = shallow(<FormInput {...mockProps}/>);
    });

    test('should render formInput component', () =>{
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should fire handler for onChange events', () =>{
        wrapper.find('input').simulate('change');
        expect(mockHandler).toHaveBeenCalled();
    });
    
    test('should render label if there is a label prop', () =>{
        expect(wrapper.exists('label')).toBe(true);
    });
    
    test('should not render label if there is not a label prop', () =>{
    
        const mockNewProps = {
            type: 'email',
            name: 'email',
            value: 'myemail@gmail.com',
            handleChange: mockHandler,
        }
    
        const newWrapper = shallow(<FormInput {...mockNewProps}/>);
    
        expect(newWrapper.exists('label')).toBe(false);
    });
});