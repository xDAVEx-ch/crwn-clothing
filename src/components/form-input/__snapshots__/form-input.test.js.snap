// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput component should render formInput component 1`] = `ShallowWrapper {}`;

exports[`should render formInput component 1`] = `ShallowWrapper {}`;
